src/Sylius/Component/Core/Test/Factory/TestPromotionFactoryInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Test\Factory;
+
+use Sylius\Component\Core\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\ActionInterface;
+use Sylius\Component\Promotion\Model\RuleInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface TestPromotionFactoryInterface
+{
+    /**
+     * @param string $name
+     *
+     * @return PromotionInterface
+     */
+    public function createPromotion($name);
+
+    /**
+     * @param string $discount
+     * @param PromotionInterface $promotion
+     *
+     * @return ActionInterface
```

Comment:
This is a good question, *de facto* about interface/class name.
My point was, that I don't see a need of separate test factories for promotions **and** actions **and** rules. They are highly connected with each other and IMO they should be created in the same service.
So the problem is - how such a service should be named? :smile: